pkg/store/proxy.go

```diff
@@ -322,6 +327,20 @@ func (s *ProxyStore) Series(originalRequest *storepb.SeriesRequest, srv storepb.
 			return status.Error(codes.Aborted, resp.GetWarning())
 		}
 
+		// For backwards compatibility, we need to make sure that each store sends response hints indicating whether
```

Comment:
Indeed this is a bit complex (cognitive load, plus hot path) for something that is always true once someone do the rollout of all components 🤔

Reply:
First of all we could add TODO to remove it in some Thanos version? 🤔